### PR TITLE
Fix wrong log when push to a new branch

### LINKED ISSUE #1111: Push success but display an error log
Don't know how to reproduce that, maybe some special commit message and I'm finding it.
```
core git:(master) git push origin master
Counting objects: 4, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (4/4), done.
Writing objects: 100% (4/4), 1.83 KiB | 0 bytes/s, done.
Total 4 (delta 1), reused 0 (delta 0)
remote: 2017/03/03 22:17:40 [...tea/models/update.go:147 PushUpdate()] [E] updateIssuesCommit: No issue number specified
To http://192.168.0.103:3000/lunny/core.git
   8e2e75f..c0ba85e  master -> master
```

### PR DESCRIPTION
**Pull request by lunny**
fix #1111

### CODE REVIEW

File: models/action.go [hunk 1]
```diff
@@ -360,7 +360,7 @@ func UpdateIssuesCommit(doer *User, repo *Repository, commits []*PushCommit) err
 
 			issue, err := GetIssueByRef(ref)
 			if err != nil {
-				if IsErrIssueNotExist(err) {
+				if IsErrIssueNotExist(err) || err == errMissingIssueNumber {
 					continue
```
Comment (strk) [inline models/action.go:364]: I wonder if GetIssueByRef should just return an ErrIssueNotExist even when an issue number is missing...

**Review by strk — COMMENTED** [review]
LGTM beside the comment

**Comment by strk** [comment]
Does the LGTM bot not look into reviews ?

File: models/action.go (hunk 1 above)
Comment (lunny) [reply models/action.go:364]: Let me check could we do that.

**Comment by appleboy** [comment]
@strk L-G-T-M bot doesn't look into review comments. :(

**Comment by pgaskin** [comment]
LGTM except for @strk's comment

File: models/action.go (hunk 1 above)
Comment (lunny) [reply models/action.go:364]: `GetIssueByRef` will return many kinds of error besides `ErrIssueNotExist` and `errMissingIssueNumber`. Don't know how to fix this.

**Pull request merged**